library/src/main/java/org/apache/apex/malhar/lib/state/spillable/SpillableComplexComponentImpl.java

```diff
@@ -50,6 +51,11 @@
   @NotNull
   private SpillableIdentifierGenerator identifierGenerator;
 
+  /**
+   * need to make sure all the buckets already created when setup.
+   */
+  protected transient Set<Long> bucketIds = Sets.newHashSet();
```

Comment:
Ids will be removed just after setup. The bucket itself is not transient, so the buckets will be created when recovery.